core/trino-parser/src/main/java/io/trino/sql/tree/NullLiteral.java

```diff
@@ -18,6 +18,8 @@
 public class NullLiteral
         extends Literal
 {
+    public static final NullLiteral NULL_LITERAL = new NullLiteral();
```

Comment:
It's not safe to reuse expression components, since analyzer binds information on per-identity basis (eg coercions).